gapis/api/vulkan/find_issues.go

```diff
@@ -55,12 +61,78 @@ func (t *findIssues) reportTo(r replay.Result) { t.res = append(t.res, r) }
 
 func (t *findIssues) Transform(ctx context.Context, id api.CmdID, cmd api.Cmd, out transform.Writer) {
 	ctx = log.Enter(ctx, "findIssues")
+	cb := CommandBuilder{Thread: cmd.Thread(), Arena: t.state.Arena}
+
 	mutateErr := cmd.Mutate(ctx, id, t.state, nil /* no builder */)
 	if mutateErr != nil {
 		// Ignore since downstream transform layers can only consume valid commands
 		return
 	}
+	// Before an instance is to be destroyed, check if it has debug report callback
+	// created by us, if so, destory the call back handle.
+	if di, ok := cmd.(*VkDestroyInstance); ok {
+		inst := di.Instance()
+		if ch, ok := t.reportCallbacks[inst]; ok {
+			out.MutateAndWrite(ctx, api.CmdNoID, cb.ReplayDestroyVkDebugReportCallback(inst, ch))
+		}
+	}
+
 	out.MutateAndWrite(ctx, id, cmd)
+
+	// After an instance is created, try to create a debug report call back handle
+	// for it. The create info is not completed, the device side code should complete
+	// the create info before calling the underlying Vulkan command.
+	if ci, ok := cmd.(*VkCreateInstance); ok {
+		inst := ci.PInstance().MustRead(ctx, cmd, out.State(), nil)
+		callbackHandle := VkDebugReportCallbackEXT(newUnusedID(true, func(x uint64) bool {
+			for _, cb := range t.reportCallbacks {
+				if uint64(cb) == x {
+					return true
+				}
+			}
+			return false
+		}))
+		callbackHandleData := out.State().AllocDataOrPanic(ctx, callbackHandle)
+		callbackCreateInfo := out.State().AllocDataOrPanic(
+			ctx, NewVkDebugReportCallbackCreateInfoEXT(
+				out.State().Arena,
+				VkStructureType_VK_STRUCTURE_TYPE_DEBUG_REPORT_CREATE_INFO_EXT, // sType
+				0, // pNext
+				VkDebugReportFlagsEXT((VkDebugReportFlagBitsEXT_VK_DEBUG_REPORT_DEBUG_BIT_EXT<<1)-1), // flags
+				0, // pfnCallback
+				0, // pUserData
+			))
+		out.MutateAndWrite(ctx, api.CmdNoID, cb.ReplayCreateVkDebugReportCallback(
+			inst,
+			callbackCreateInfo.Ptr(),
+			callbackHandleData.Ptr(),
+			true,
+		).AddRead(
+			callbackCreateInfo.Data(),
+		).AddWrite(
+			callbackHandleData.Data(),
+		))
+	}
+
+	if !t.reportHandlerSet {
+		out.MutateAndWrite(ctx, api.CmdNoID, cb.Custom(func(ctx context.Context, s *api.GlobalState, b *builder.Builder) error {
+			b.RegisterNotificationReader(func(n gapir.Notification) {
+				vkApi := API{}
+				if uint8(n.GetApiIndex()) != vkApi.Index() {
+					return
+				}
+				var issue replay.Issue
+				// TODO: For MEC, we need to minus the number of initial commands
+				// TODO: Figure out what to do for commands inserted by GAPID, whose label is ~0.
+				issue.Command = api.CmdID(n.GetLabel())
+				issue.Severity = service.Severity(uint32(n.GetSeverity()))
```

Comment:
Do we have a better way to enforce this? I found in gapis/service/service.proto, and core/log, we also just note in the comments to say their value should be the same.